src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -399,15 +399,15 @@ internal bool CheckValueKind(SyntaxNode node, BoundExpression expr, BindValueKin
                     return CheckPropertyValueKind(node, expr, valueKind, checkingReceiver, diagnostics);
 
                 case BoundKind.IndexOrRangePatternIndexerAccess:
-                    var patternIndexer = ((BoundIndexOrRangePatternIndexerAccess)expr);
-                    if (patternIndexer.PatternSymbol.Kind == SymbolKind.Property)
+                    var implicitIndexer = (BoundIndexOrRangePatternIndexerAccess)expr;
+                    if (implicitIndexer.IndexerAccess is BoundIndexerAccess implicitIndexerAccess)
                     {
                         // If this is an Index indexer, PatternSymbol should be a property, pointing to the
-                        // pattern indexer. If it's a Range access, it will be a method, pointing to a Slice method
+                        // implicit indexer. If it's a Range access, it will be a method, pointing to a Slice method
                         // and it's handled below as part of invocations.
-                        return CheckPropertyValueKind(node, expr, valueKind, checkingReceiver, diagnostics);
+                        return CheckValueKind(node, implicitIndexerAccess, valueKind, checkingReceiver, diagnostics);
                     }
-                    Debug.Assert(patternIndexer.PatternSymbol.Kind == SymbolKind.Method);
+                    Debug.Assert(implicitIndexer.IndexerAccess is BoundCall);
```

Comment:
>Debug.Assert(implicitIndexer.IndexerAccess is BoundCall); [](http://example.com/codeflow?start=20&length=57)

Wouldn't always delegating to the underlying IndexerAccess always work. Is there a real need to this split?